internal/xpkg/fetch.go

```diff
@@ -38,14 +44,46 @@ type Fetcher interface {
 type K8sFetcher struct {
 	client    kubernetes.Interface
 	namespace string
+	transport http.RoundTripper
 }
 
 // NewK8sFetcher creates a new K8sFetcher.
-func NewK8sFetcher(client kubernetes.Interface, namespace string) *K8sFetcher {
+func NewK8sFetcher(client kubernetes.Interface, namespace, caBundlePath string) (*K8sFetcher, error) {
+
+	// return early with default transport if no additional CAs
+	if caBundlePath == "" {
+		return &K8sFetcher{
+			client:    client,
+			namespace: namespace,
+			transport: http.DefaultTransport,
+		}, nil
+	}
+
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	// Read in the cert file
+	certs, err := ioutil.ReadFile(filepath.Clean(caBundlePath))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to append %q to RootCAs", caBundlePath)
+	}
+
+	// Append our cert to the system pool
+	if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+		return nil, errors.Errorf("No certificates could be parsed from %q", caBundlePath)
+	}
+
+	var customCATransport http.RoundTripper = http.DefaultTransport
+	customCATransport.(*http.Transport).TLSClientConfig = &tls.Config{RootCAs: rootCAs, MinVersion: tls.VersionTLS12}
```

Comment:
@zonybob while I think encountering it would be an edge case (because hopefully any registry being used is on at least TLS 1.2), I am somewhat inclined to just fallback to default TLS version min defined in `tls` pkg (which is currently `1.0`). Do you see a compelling reason to not leave this to std library?